natvis/object_visualizer.cpp

```diff
@@ -424,89 +375,85 @@ void GetInterfaceData(
                     {
-                        propCategory = PropertyCategory::Guid;
+                        if (name == "Matrix3x2") { name = "float3x2"; }
+                        else if (name == "Matrix4x4") { name = "float4x4"; }
+                        else if (name == "Plane") { name = "plane"; }
+                        else if (name == "Quaternion") { name = "quaternion"; }
+                        else if (name == "Vector2") { name = "float2"; }
+                        else if (name == "Vector3") { name = "float3"; }
+                        else if (name == "Vector4") { name = "float4"; }
                     }
-                    else
-                    {
-                        auto ns = std::string(type.TypeNamespace());
-                        auto name = std::string(type.TypeName());
-
-                        // Map numeric type names
-                        if (ns == "Windows.Foundation.Numerics")
-                        {
-                            if (name == "Matrix3x2") { name = "float3x2"; }
-                            else if (name == "Matrix4x4") { name = "float4x4"; }
-                            else if (name == "Plane") { name = "plane"; }
-                            else if (name == "Quaternion") { name = "quaternion"; }
-                            else if (name == "Vector2") { name = "float2"; }
-                            else if (name == "Vector3") { name = "float3"; }
-                            else if (name == "Vector4") { name = "float4"; }
-                        }
 
-                        // Types come back from winmd files with '.', need to be '::'
-                        // Ex. Windows.Foundation.Uri needs to be Windows::Foundation::Uri
-                        auto fullTypeName = ns + "::" + name;
-                        wchar_t cppTypename[500];
-                        size_t i, j;
-                        for (i = 0, j = 0; i < (fullTypeName.length() + 1); i++, j++)
-                        {
-                            if (fullTypeName[i] == L'.')
-                            {
-                                cppTypename[j++] = L':';
-                                cppTypename[j] = L':';
-                            }
-                            else
-                            {
-                                cppTypename[j] = fullTypeName[i];
-                            }
-                        }
-
-                        propDisplayType = std::wstring(L"winrt::") + cppTypename;
-                        if(get_category(type) == category::class_type)
+                    // Types come back from winmd files with '.', need to be '::'
+                    // Ex. Windows.Foundation.Uri needs to be Windows::Foundation::Uri
+                    auto fullTypeName = ns + "::" + name;
+                    wchar_t cppTypename[500];
+                    size_t i, j;
+                    for (i = 0, j = 0; i < (fullTypeName.length() + 1); i++, j++)
```

Comment:
Would this this loop not be a lot less wonky if `cppTypename` were a `std::wstring`?

Reply:
is there a simpler way to replace all occurrences of "." with "::" in a std::wstring?

Reply:
No. :)

I just mean, it might be simpler as follows:

```C++
std::wstring transform(std::string_view source)
{
    std::wstring destination;

    for (auto c : source)
    {
        if (c == '.')
        {
            destination += L"::";
        }
        else
        {
            destination += c;
        }
    }

    return destination;
}
```

No raw buffers and wonky indexes to manage.

Reply:
in exchange for lots of reallocations, but yes looks nicer

Reply:
https://en.cppreference.com/w/cpp/string/basic_string/reserve